cfme/common/provider.py

```diff
@@ -50,6 +52,7 @@ class BaseProvider(Taggable, Updateable, SummaryMixin):
     _properties_region = None
     add_provider_button = None
     save_button = None
+    appliance = get_or_create_current_appliance()
```

Comment:
So, this has to be ```appliance = CurrentAppliance()``` else it will set it to be one appliance and never change with the stack. The idea here with the class based attribute is that ```appliance``` attribute will change with the stack. As there is no instance if you are accessing the attribute on the class, we absolutely want it to always return whichever appliance is the **current** one.